src/main/java/uk/gov/pay/directdebit/healthcheck/resources/HealthCheckResource.java

```diff
@@ -33,28 +33,24 @@ public HealthCheckResource(Environment environment) {
     @Produces(APPLICATION_JSON)
     @Timed
     public Response healthCheck() {
-        SortedMap<String, HealthCheck.Result> results = environment.healthChecks().runHealthChecks();
-
-        Map<String, Map<String, Object>> response = getResponse(results);
-
-        boolean healthy = results.size() == results.values()
+        Map<String, Map<String, Object>> results = environment
+                .healthChecks()
+                .runHealthChecks()
+                .entrySet()
                 .stream()
-                .filter(HealthCheck.Result::isHealthy)
-                .count();
-
-        if (healthy) {
-            return Response.ok().entity(response).build();
-        }
-        return status(503).entity(response).build();
+                .collect(Collectors.toMap(Map.Entry::getKey,
+                        healthCheckResult -> ImmutableMap.of(
+                                "healthy", healthCheckResult.getValue().isHealthy(),
+                                "message", defaultString(healthCheckResult.getValue().getMessage(), "Healthy")
+                        )
+                ));
+
+        return Response.status(allHealthy(results.values()) ? OK : SERVICE_UNAVAILABLE)
+                .entity(results)
+                .build();
     }
 
-    private Map<String, Map<String, Object>> getResponse(SortedMap<String, HealthCheck.Result> results) {
-        Map<String, Map<String, Object>> response = new HashMap<>();
-        for (SortedMap.Entry<String, HealthCheck.Result> entry : results.entrySet()) {
-            response.put(entry.getKey(), ImmutableMap.of(
-                    "healthy", entry.getValue().isHealthy(),
-                    "message", isBlank(entry.getValue().getMessage()) ? "Healthy" : entry.getValue().getMessage()));
-        }
-        return response;
+    private boolean allHealthy(Collection<Map<String, Object>> results) {
+        return results.stream().allMatch(map -> (boolean) map.get("healthy"));
```

Comment:
This is a result of my meddling and over zealous behaviour having read the latest Refactoring book. I felt that a local variable to hold the health check results was unnecessary since it was immediately transformed into the `results` (the later `(boolean)` was the lesser of two evils). However if you feel this hinders comprehension then that defeats part of the purpose and I hope @rhowe-gds can find it in his heart to forgive me.

Reply:
How about we just rewrite this in lisp?

Reply:
Here's a revised attempt - it's not lisp yet